### PR TITLE
[iov-tendermint-rpc] Fix bug in situations where listen and execute is used

### PR DESCRIPTION
**Pull request by webmaster128**

### CODE REVIEW

File: packages/iov-tendermint-rpc/src/rpcclient.spec.ts [hunk 1]
```diff
@@ -112,6 +112,41 @@ describe("RpcClient", () => {
       });
     });
 
+    it("can execute commands while listening to events", done => {
+      pendingWithoutTendermint();
+
+      const client = new WebsocketClient(tendermintUrl);
+
+      const query = "tm.event='NewBlockHeader'";
```
Comment (ethanfrey) [inline packages/iov-tendermint-rpc/src/rpcclient.spec.ts]: Nice low-level setup

File: packages/iov-tendermint-rpc/src/rpcclient.spec.ts [hunk 2]
```diff
@@ -112,6 +112,41 @@ describe("RpcClient", () => {
...
+            // wait 1.5s and check we did not get more events
+            setTimeout(() => {
+              expect(events.length).toEqual(2);
+              done();
+            }, 1500);
+          }
+        },
+      });
+
+      const startusResponse = client.execute(jsonRpcWith(Method.STATUS));
```
Comment (ethanfrey) [inline packages/iov-tendermint-rpc/src/rpcclient.spec.ts]: Good test, but i would almost put some delay here to make sure that the subscription finished first.

`setTimeout(100, () => ...)` ?

File: packages/iov-tendermint-rpc/src/rpcclient.ts [hunk 3]
```diff
@@ -185,16 +185,31 @@ export class WebsocketClient implements RpcStreamingClient {
       // only one of the two listeners should fire, and it will
       // deregister the other so as not to cause a leak.
 
+      const good = {
+        eventName: id,
+        listener: (data: JsonRpcResponse) => {
+          unregisterGoodAndBad();
```
Comment (ethanfrey) [inline packages/iov-tendermint-rpc/src/rpcclient.ts]: maybe just `unregisterBad()`? `unregisterGood` would already happen via `once`

Comment (webmaster128) [reply packages/iov-tendermint-rpc/src/rpcclient.ts]: You mean having two different cleanup functions? Right now the idea is to cleanup both such that only one function is needed.

File: packages/iov-tendermint-rpc/src/rpcclient.spec.ts (hunk 2 above)
Comment (webmaster128) [reply packages/iov-tendermint-rpc/src/rpcclient.spec.ts]: could be done, yes. My idea was to keep the test as simple as possible while catching the bug. But I am open for future improvements

**Pull request merged**